商品削除機能

ken-forest opened the pull request:
# What
１度出した商品を削除する機能を作る

# Why
出品者は時には、一度出した商品を無かったことにしたい場合がある。
そういう時に、商品を削除する機能が必要である。
これがないと、出品者に不利益を与えてしまう可能性があるので
この機能は必須機能であると思われる。

＜Gyazoで撮影したもの＞
・ログイン状態の出品者のみ、詳細ページの削除ボタンを押すと、出品した商品を削除できる
①ユーザー「ken」がトップページ→オムライスの絵画ページへ
https://gyazo.com/c85ea47866afa571b02903f38f2ce24b

②削除を実行→トップページ→ページの商品一覧にオムライスの絵画が無いことを確認
https://gyazo.com/f346b0a16e9d9c012b2f9c6ba29ddaa2

ken-forest committed:
itemsコントローラに、destroyアクションを定義(中身は空)

ken-forest committed:
詳細ページの削除ボタンのパスを「item_path(@item.id)」に変更

ken-forest committed:
itemsコントローラのdestroyアクション内を記述

app/controllers/items_controller.rbのdestroyアクションに以下を記述
item = Item.find(params[:id])
item.destroy
redirect_to root_path
、、商品を削除したあとは
トップページに戻るようにした

ken-forest commented:
レビューをよろしくお願いします！

shodai-ito commented on app/controllers/items_controller.rb:
```diff
@@ -39,6 +39,13 @@ def update
   end
 
+  def destroy
+    item = Item.find(params[:id])
+    item.destroy
```
こちら変数を@itemとすることで、define_itemメソッドを適用できます。
コードも短くなるため、是非利用しましょう！

shodai-ito reviewed (commented):
レビュー依頼ありがとうございます！
１点だけコメントしましたので、ご確認をお願いいたします！

その後、再度レビュー依頼フォームのご提出をよろしくお願いいたします。

ken-forest committed:
itemsコントローラのdestroyアクションの記述を変更

＜define_itemメソッドを使用＞
・item = Item.find(params[:id])を削除
・itemをインスタンス変数@itemに変更
・before_actionのdefine_itemメソッドの発動箇所にdestroyアクションも追加

ken-forest commented:
ご指摘いただいたdestroyアクションの記述を変更しました。
コード変更後も削除はしっかり発動してました。
＜Gyazoにて動作確認してます＞
https://gyazo.com/5d93e4b9bdbfcf5d7f24826d4f5f7d45

引き続き、レビューをよろしくお願いします！

zonodance commented on app/controllers/items_controller.rb:
```diff
@@ -39,6 +39,13 @@ def update
+    
+    @item.destroy
+    redirect_to root_path
+
```
「出品者と今ログインしているユーザーが同じ」場合のみ削除できるような条件分岐をdestroyアクションに追加してあげましょう！

zonodance reviewed (commented):
レビュー依頼ありがとうございます！
1点コメントをさせていただきましたので修正の程よろしくお願いいたします。

ken-forest committed:
before_actionのmove_to_top_pageの発動箇所にdestroyアクションを追加

これにより、destroyアクションが発動する前に、、
unless current_user == @item.user
      redirect_to root_path
 end
、、の条件がかけられる事になる。

ken-forest commented:
ご指摘の箇所を１つ、修正しました。
引き続き、レビューをよろしくお願いします！

zonodance reviewed (commented):
レビュー依頼ありがとうございます！
また、毎回ご丁寧にコメントもいただきありがとうございます。

LGTMとさせていただきたいです！
商品削除機能の実装お疲れさまでした！
次の機能に進む前に忘れずにmergeしておきましょう！
